### DISCUSSION TITLE
Play content via Kodi Issue and YT player suggestion

### DISCUSSION THREAD

**Discussion by DudeOnInternet** [TeamNewPipe/NewPipe#10269] [Questions]
Hello all,
I do not know whether this is the right place to start this topic.
In case it is not forgive me as I'm totally new in this open source SW development community.....
Since a few months my YouTube player failed on my RPI with Kodi and what appears to be the problem that YT is using new API functionality which causes my YT player not to play videos anymore, which I steamed using NewPipe, hence I start the discussion here.
I've looked all over internet for a while now to find a solution which would not require me to make a personal API key, as that is basically only given as universal solution. 
Also looked at an alternative Kodi YT player but this also requires a personal API to be used.
This is not a solution for me as I want to stay anonymous, especially for Google.
This is also the idea behind NewTube, I guess.
Is there perhaps already a solution to circumvent the API issue, which you as insiders in the community might know, but is (understandingly) not broadly published online?
Are there perhaps dummie API's which can be used, I guess that is also what NewPipe has to do to get access to the YT servers?
If no solution is yet in place would it be an idea for you developers to make a Kodi player, to play YT video's without requiring personal API (like done for NewPipe), and than also to add support Rumble and Odysee and even more YT alternatives, as ideally we have to also start moving away from YT?

For now thanks and looking forward to any comments!!!

**Comment by opusforlife2** [comment]
You're looking for https://github.com/TeamNewPipe/NewPipe/issues/new/choose

**Reply by DudeOnInternet** [reply]
Thank you for sharing this link, but that is exactly where I came from. 
There are 4 subjects which I could use, and I took the " ❓ Question Ask about anything NewPipe-related" option as that suited my question the most, as my question is not a bug report, not (yet) a feature request and definitely not a security vulnerability.
As I did not get a answer on whether there already is a fix/workaround for it, I think it is a bit to early to make it a feature request.
I was hoping on some sort of discussion, and perhaps also getting some suggestions/tips.
Can you perhaps elaborate a bit more on what is expected on how to proceed, as for now, by only sending me that link, that is not clear to me at all.

Thanks!!

**Reply by opusforlife2** [reply]
> which causes my YT player not to play videos anymore, which I steamed using NewPipe

If something worked for you before, but it doesn't work now, that's a bug. Hence you open a bug report.

**Reply by DudeOnInternet** [reply]
The suggestion to create an bug issue, I do not think is the correct way to go.
This as said in my initial message, the issue does not seem to be related to the NewPipe App itself, but to the youtube API requirement the YT player has for Kodi. I can not play any YT movies with the app itself anymore.
So for me I do not think this is a bug on your side, hence my question: Is there a way to circumvent the API requirement?
I do not think I'm the only one facing this problem, so how does the NP community tackle this issue?
As this discussion has been closed very quickly, there was no chance for a discussion to get started on the possibilities.
Still your advise to create bug issue, or any new thoughts?

And then additionally  made a suggestion to solve it differently is to make an specific NewPipe YT player app for Kodi.
Should I than also create a feature request for this than?

**Comment by opusforlife2** [comment]
> the issue does not seem to be related to the NewPipe App itself, but to the youtube API requirement the YT player has for Kodi. I can not play any YT movies with the app itself anymore

Please clarify some things:

1. What exact steps did you use which involved Newpipe?
2. Where in these steps is the failure occurring?

**Reply by DudeOnInternet** [reply]
I use NewPipe to follow different YT channels, and store my favorites.
Whenever I want to watch a video when being in my living room, I use the "Play with Kodi" feature.
This will first activate the Kore Kodi remote control app on my phone.
This normally than automatically start the YT app on kodi, and then I can watch the selected video on my large screen television.
This used to work till lately, in the mean time say 3 months ago, now nothing happens anymore on Kodi.
When I now press "Play with Kodi" this still activates the Kodi remote control app on my phone, and than nothing happens any more on Kodi.
I have tried Kodi on my RPI4 and Raspbian OS, and I have tried Kodi on an Odroid N2+ using CoreELEC, and on both it is failing.

**Reply by opusforlife2** [reply]
> This normally than automatically start the YT app on kodi, and then I can watch the selected video on my large screen television.
This used to work till lately, in the mean time say 3 months ago, now nothing happens anymore on Kodi.

Thanks. This is the bug you need to create a new issue for. Go ahead.
